src/refdb_fs.c

```diff
@@ -1485,6 +1517,17 @@ static int refdb_fs_backend__delete_tail(
 		goto cleanup;
 	}
 
+	if ((flags & GIT_REFDB__SKIP_REFLOG) == 0) {
+		if ((error = should_write_reflog(&should_write, backend->repo, ref->name)) < 0)
+			goto cleanup;
+
+		if (should_write) {
+			const git_oid *old_id = git_reference_target(ref);
+			if ((error = maybe_append_head(backend, ref, old_id, NULL, who, message)) < 0)
+				goto cleanup;
+		}
+	}
```

Comment:
This hunk looks unrelated to what this commit is doing?

Reply:
Well, this is the part that actually writes to the reflog when deleting. That's taken from `write_tail`, minus the "write to our own reflog part" because we already deleted it.

I'll rewrite the commit message though, as it's not clear (must explain why, not what).